tests/send_specific_receipts.go

```diff
@@ -0,0 +1,147 @@
+	// Check messge delivered
+	delivered, err := subnetBTeleporterMessenger.MessageReceived(&bind.CallOpts{}, subnetAInfo.BlockchainID, messageID1)
+	Expect(err).Should(BeNil())
+	Expect(delivered).Should(BeTrue())
+
+	// Send second message from Subnet A to Subnet B with fee amount 5
+	sendCrossChainMsgReceipt, messageID2 := utils.SendCrossChainMessageAndWaitForAcceptance(
+		ctx, subnetAInfo, subnetBInfo, sendCrossChainMessageInput, fundedAddress, fundedKey, subnetATeleporterMessenger)
+
+	// Relay message from SubnetA to SubnetB
+	network.RelayMessage(ctx, sendCrossChainMsgReceipt, subnetAInfo, subnetBInfo, false, true)
+	// Check delivered
+	delivered, err = subnetBTeleporterMessenger.MessageReceived(&bind.CallOpts{}, subnetAInfo.BlockchainID, messageID2)
+	Expect(err).Should(BeNil())
+	Expect(delivered).Should(BeTrue())
+
+	// Relayer send specific receipts to get reward of relaying two messages
+	receipt, messageID := utils.SendSpecifiedReceiptsAndWaitForAcceptance(
+		ctx,
+		subnetAInfo.BlockchainID,
+		subnetBInfo,
+		[]*big.Int{messageID1, messageID2},
+		teleportermessenger.TeleporterFeeInfo{
+			ContractAddress: mockTokenAddress,
+			Amount:          big.NewInt(0),
+		},
+		[]common.Address{},
+		fundedAddress,
+		fundedKey,
+		subnetBTeleporterMessenger,
+	)
+
+	// Relay message from Subnet B to Subnet A
+	network.RelayMessage(ctx, receipt, subnetBInfo, subnetAInfo, false, true)
+	// Check delivered
+	delivered, err = subnetATeleporterMessenger.MessageReceived(&bind.CallOpts{}, subnetBInfo.BlockchainID, messageID)
+	Expect(err).Should(BeNil())
+	Expect(delivered).Should(BeTrue())
+
+	// Check the reward amount. The reward amount should be 10
+	amount, err := subnetATeleporterMessenger.CheckRelayerRewardAmount(&bind.CallOpts{}, fundedAddress, mockTokenAddress)
+	Expect(err).Should(BeNil())
+	Expect(amount).Should(Equal(totalAccumulatedRelayerFee))
+
+	// Send message from Subnet B to Subnet A to trigger the "regular" method of delivering receipts.
+	// The next message from B->A will contain the same receipts that were manually sent in the above steps,
+	// but they should not be processed again on Subnet A.
+	sendCrossChainMessageInput = teleportermessenger.TeleporterMessageInput{
+		DestinationChainID: subnetAInfo.BlockchainID,
+		DestinationAddress: fundedAddress,
+		FeeInfo: teleportermessenger.TeleporterFeeInfo{
+			ContractAddress: mockTokenAddress,
+			Amount:          big.NewInt(0),
+		},
+		RequiredGasLimit:        big.NewInt(1),
+		AllowedRelayerAddresses: []common.Address{},
+		Message:                 []byte{1, 2, 3, 4},
+	}
+
+	// This message will also have the same receipt as the previous message
+	receipt, messageID = utils.SendCrossChainMessageAndWaitForAcceptance(
+		ctx, subnetBInfo, subnetAInfo, sendCrossChainMessageInput, fundedAddress, fundedKey, subnetBTeleporterMessenger)
+
+	// Relay message from Subnet B to Subnet A
+	network.RelayMessage(ctx, receipt, subnetBInfo, subnetAInfo, false, true)
```

Comment:
can we unpack the Teleporer message that we relayed here, and check that receipts include the duplicated receipt for message IDs above?